onnxruntime/core/session/inference_session.cc

```diff
@@ -528,6 +528,11 @@ common::Status InferenceSession::Initialize() {
     // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
     ORT_RETURN_IF_ERROR(graph.Resolve());
 
+    if (session_options_.graph_optimization_level < TransformerLevel::Level3 && !session_options_.optimized_model_filepath.empty()) {
```

Comment:
if model file is not empty and graph optimization level is set to 3 then add some log to indicate to the user why we are not serializing the model
In current scenario it comes off as a bug